domain/src/main/kotlin/com/krzykrucz/elesson/currentlesson/domain/topic/domain/TopicWorkflowImplementation.kt

```diff
@@ -0,0 +1,22 @@
+package com.krzykrucz.elesson.currentlesson.domain.topic.domain
+
+import arrow.core.getOrElse
+import com.krzykrucz.elesson.currentlesson.domain.NaturalNumber
+import com.krzykrucz.elesson.currentlesson.domain.startlesson.LessonIdentifier
+
+fun chooseTopic(): ChooseTopic = { topicTitle, finishedLessonsCount, checkedAttendance ->
+    val attendance = checkedAttendance.attendance
+    InProgressLesson(
+            LessonIdentifier(
+                    date = attendance.date,
+                    lessonHourNumber = attendance.lessonHourNumber,
+                    className = attendance.className
+            ),
+            LessonTopic(LessonOrdinalNumber(calculateLessonOrdinalNumber(finishedLessonsCount)), topicTitle, attendance.date),
+            checkedAttendance
+    )
+}
+
+private fun calculateLessonOrdinalNumber(finishedLessonsCount: FinishedLessonsCount) =
+        NaturalNumber.of(finishedLessonsCount.count.number + NaturalNumber.ONE.number)
```

Comment:
interesting that we model to such extends like `NaturalNumber.ONE` and similar :smile: AFAIK, in Kotlin it's possible to override `plus` operator to have this operation more natural and readable, what do you think?

Reply:
Great idea, done